doc/Docker_SDK_Cross_Compile.md

```diff
@@ -0,0 +1,231 @@
+WORKDIR azure-iot-sdk-c
+
+# Create a working directory for the cmake operations
+RUN mkdir cmake
+WORKDIR cmake
+
+# Create a cmake toolchain file on the fly
+RUN echo "SET(CMAKE_SYSTEM_NAME Linux)     # this one is important" > toolchain.cmake
+RUN echo "SET(CMAKE_SYSTEM_VERSION 1)      # this one not so much" >> toolchain.cmake
+RUN echo "SET(CMAKE_SYSROOT ${TOOLCHAIN_SYSROOT})" >> toolchain.cmake
+RUN echo "SET(CMAKE_C_COMPILER ${TOOLCHAIN_EXES}/${TOOLCHAIN_NAME}-gcc)" >> toolchain.cmake
+RUN echo "SET(CMAKE_CXX_COMPILER ${TOOLCHAIN_EXES}/${TOOLCHAIN_NAME}-g++)" >> toolchain.cmake
+RUN echo "SET(CMAKE_FIND_ROOT_PATH $ENV{TOOLCHAIN_SYSROOT})" >> toolchain.cmake
+RUN echo "SET(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)" >> toolchain.cmake
+RUN echo "SET(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)" >> toolchain.cmake
+RUN echo "SET(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)" >> toolchain.cmake
+RUN echo "SET(set_trusted_cert_in_samples true CACHE BOOL \"Force use of TrustedCerts option\" FORCE)" >> toolchain.cmake
+
+# Build the SDK. This will use the OpenSSL, cURL and uuid binaries that we built before
+RUN cmake -DCMAKE_TOOLCHAIN_FILE=toolchain.cmake -DCMAKE_INSTALL_PREFIX=${TOOLCHAIN_PREFIX} ..
+RUN make
+RUN make install
+
+# Finally a sanity check to make sure the files are there
+RUN ls -al ${TOOLCHAIN_PREFIX}/lib
+RUN ls -al ${TOOLCHAIN_PREFIX}/include
+
+# Go to project root
+WORKDIR ../..
+```
+
+To run this script save it to a directory and change to that directory then simply enter:
+```bash
+docker build -t mipsbuild:latest . --network=host
+```
+You can replace the value 'mipsbuild' with any name that describes your build. In this instance it is building for a MIPS32 processor hence the name. Once the build is complete you will be ready to build your application against the libraries just built.
+
+*Note:* The build of the Azure IoT C SDK uses cmake to generate the makefiles. In order to instruct cmake to perform a cross-compile one uses a cmake toolchain file. This tells cmake where to find the libraries etc. in the toolchain rather than on the host. This file is created on the fly in the above script to reduce dependencies on external files. Alternatively one could create this file outside the script and import into the container.
+## Building Your Application
+Once you have successfully built the SDK you are now ready to create your application. Your application will also need to be cross compiled so adding additional steps to the end of the Dockerfile script to build your code is likely the easiest way to accomplish this. If you set up your application to also build with cmake then you can simply use the cmake toolchain file created to build the SDK. 
+
+### Sample ```CMakeLists.txt``` for an Application
+Here is a simple ```CMakeLists.txt``` file that demonstrates how to build an application that uses the libraries built above. This sample, for demonstration purposes uses the ```iothub_convenience_sample.c``` sample from the SDK.
```

Comment:
>Here is a simple ```CMakeLists.txt`` [](start = 0, length = 36)

It is not clear where the CMakeLists.txt starts and ends. Probably a tabulation would help?

Reply:
If you view the file in a markup file renderer you'll see it is obvious since it is fenced by ``` cmake which tabs it in slightly and does syntax highlighting. [https://github.com/Azure/azure-iot-sdk-c/blob/mrdockercc/doc/Docker_SDK_Cross_Compile.md](https://github.com/Azure/azure-iot-sdk-c/blob/mrdockercc/doc/Docker_SDK_Cross_Compile.md) 

---
In reply to: [213761474](https://github.com/Azure/azure-iot-sdk-c/pull/608#discussion_r213761474) [](ancestors = 213761474)